TechnicalDocuments/TechnicalDebt.md

```diff
@@ -0,0 +1,115 @@
+
+## Objective-C code and GatewayManager.
+
+GatewayManager supplies global dependencies to legacy Objective-C code. There was plenty of code in version one and two of the Babylon app that relied heavily on singletons.
+
+| Objective-C view Controllers | Location | Comments |
+| ---------------------------- | ---------| -------- |
+| BillingInformationViewController | Clinical Records | |
+| EditMembershipCodeViewController | Clinical Records | |
+| MembershipTableViewController | Me Tab | |
+| SurveyViewController | Clinical Records | |
+| InsuranceViewController | Clinical Records | |
+| BBAddAdditionalPatientInformationViewControllerV2 | Clinical Records | Might be dead code |
+
+## Free Standing Swift View Controllers that should be Moved to Bento.
+
+| View Controller | Comments |
+| --------------- | -------- |
+| InfoViewController | Could be replaced with a Bento Component for scrolling through a couple of info screens. Only used for the GP @ hand on-boarding so could be moved from BabylonUI. |
+| SegmentedViewController | Currently only used in Health Check to show the the two bands with organs and layers. Should probably be moved from BabylonUI. |
+| IntroViewController | Could potentially be merged with InfoViewController. |
+
+## View Models and Controllers that use Forms V2.
+
+Forms V2 is no longer maintained and we should refactor these view controllers to use Bento.
+Porting these view controllers to Bento will also make it easier to apply an app wide style guide (design library).
+
+| View Model | Comments |
+| ---------- | -------- |
+| AppointmentCancellationViewModel | |
+| AppointmentConsultationNotesViewModel | |
+| AppointmentDetailsViewModel | |
+| AppointmentNotesViewModel | |
+| AppointmentPrescriptionViewModel | CE-30 |
+| AppointmentReferralViewModel | |
+| AppointmentReferralsViewModel | |
+| AppointmentReplayViewModel | |
+| AppointmentListViewModel | |
+| BookAppointmentViewModel | |
+| PractitionerDetailsViewModel | |
+| ChatHistoryViewModel | |
+| MedicalHistoryViewModel | |
+| PersonalDetailsViewModel | |
+| InfoItemsViewModel | |
+| GPDetailsViewModel | |
+| AddAddressViewModel | |
+| PrivacySettingsViewModel | |
+| ChooseCountryViewModel | |
+| ForgotPasswordViewModel | |
+| NotificationsViewModel | |
+| ProfileViewModel | |
+| SignInViewModel | |
+| NHSRegistrationStep1ViewController | |
+| NHSRegistrationStep2ViewController | |
+
+## Accessing Business Controllers Directly instead of using the SDK.
+
+Some of these business controllers are not defined inside the SDK. Most of them will be moved to the SDK, but some of them will stay in the Babylon project. Once the first, complete version of the SDK has been published there should not be any business controllers left in BabylonCore. Business controllers that are defined outside the SDK should have a documenting comment that explain why they are not part of the SDK.
```

Comment:
This should mention that we must use all our SDK's.

Reply:
I am filling in a lot more detail about this in a document that I have started to write today.